From: Felipe Sateler <fsateler@debian.org>
Date: Thu, 13 Apr 2017 19:33:59 -0300
Subject: machinectl: make sure that inability to get OS version isn't
 consider fatal for machine list (#4972)

Fixes: #4918
(cherry picked from commit 4b2419165ce409ee55ce96a926302f89685f2293)
---
 src/machine/machinectl.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/machine/machinectl.c b/src/machine/machinectl.c
index 7b9be3b..c2ab202 100644
--- a/src/machine/machinectl.c
+++ b/src/machine/machinectl.c
@@ -242,23 +242,18 @@ static int list_machines(int argc, char *argv[], void *userdata) {
                 if (name[0] == '.' && !arg_all)
                         continue;
 
-                if (!GREEDY_REALLOC(machines, n_allocated, n_machines + 1)) {
+                if (!GREEDY_REALLOC0(machines, n_allocated, n_machines + 1)) {
                         r = log_oom();
                         goto out;
                 }
 
-                machines[n_machines].os = NULL;
-                machines[n_machines].version_id = NULL;
+                machines[n_machines].name = name;
+                machines[n_machines].class = class;
+                machines[n_machines].service = service;
                 r = get_os_release_property(bus, name,
                                 "ID\0" "VERSION_ID\0",
                                 &machines[n_machines].os,
                                 &machines[n_machines].version_id);
-                if (r < 0)
-                        goto out;
-
-                machines[n_machines].name = name;
-                machines[n_machines].class = class;
-                machines[n_machines].service = service;
 
                 l = strlen(name);
                 if (l > max_name)
